parsers/CO.py

```diff
@@ -1,45 +1,246 @@
 #!/usr/bin/env python3
 import json
-from datetime import datetime
+from datetime import datetime, timedelta
 from logging import Logger, getLogger
-from typing import Optional
+from typing import Any, Dict, List, Optional
 
 import arrow
+from pydataxm import ReadDB
 from requests import Session
 
+from electricitymap.contrib.config import ZoneKey
+from electricitymap.contrib.lib.models.event_lists import (
+    PriceList,
+    ProductionBreakdownList,
+    TotalConsumptionList,
+)
+from electricitymap.contrib.lib.models.events import ProductionMix
+from parsers.lib.config import refetch_frequency
+from parsers.lib.exceptions import ParserException
+
 #   PARSER FOR COLOMBIA / DEMAND-ONLY as of 2023-02-11 / 5-minute-granularity / returns demand data of the recent day
 #   MAIN_WEBSITE = https://www.xm.com.co/consumo/demanda-en-tiempo-real
 colombia_demand_URL = "https://serviciosfacturacion.xm.com.co/XM.Portal.Indicadores/api/Operacion/DemandaTiempoReal"
 
 TZ = "America/Bogota"  # UTC-5
-demand_list = []
 
+# dictionnary of plant types XM to EMaps https://www.xm.com.co/generacion/tipos
+PRODUCTION_MAPPING = {
+    "AGUA": "hydro",
+    "BIOGAS": "biomass",
+    "CARBON": "coal",
+    "GAS": "gas",
+    "BAGAZO": "biomass",
+    "RAD SOLAR": "solar",
+    "VIENTO": "wind",
+    "COMBUSTOLEO": "oil",
+    "ACPM": "oil",
+    "JET-A1": "oil",
+}
+
+XM_DELAY = 2
 
+
+@refetch_frequency(timedelta(days=1))
 def fetch_consumption(
     zone_key: str = "CO",
     session: Session = Session(),
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
-):
-    if target_datetime:
-        raise NotImplementedError("This parser is not yet able to parse past dates.")
-    with session.get(colombia_demand_URL, verify=False) as response:
-        demand_json = json.loads(response.content)
-        demand_data = demand_json["Variables"][0]["Datos"]
-
-        for datapoint in demand_data:
+) -> List[Dict[str, Any]]:
+    if target_datetime is None:
+        return fetch_live_consumption(zone_key, session, target_datetime, logger)
+    else:
+        return fetch_historical_consumption(zone_key, session, target_datetime, logger)
+
+
+def fetch_live_consumption(
+    zone_key: str = "CO",
```

Comment:
```suggestion
    zone_key: ZoneKey,
```
See suggestion for `fetch_historical_consumption`